llama_index/indices/struct_store/json.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Optional, Sequence, Union, List, Dict
+
+from llama_index.data_structs.node import Node
+from llama_index.data_structs.table import JSONStructDatapoint
+from llama_index.indices.base_retriever import BaseRetriever
+from llama_index.indices.query.base import BaseQueryEngine
+from llama_index.indices.struct_store.base import BaseGPTStructStoreIndex
+
+JSONType = Union[Dict[str, "JSONType"], List["JSONType"], str, int, float, bool, None]
+
+
+class GPTJSONIndex(BaseGPTStructStoreIndex[JSONStructDatapoint]):
```

Comment:
I think a meta question here (for everyone), should we even call these "index"?
It barely fits with the `Index` interface as it stands.

Reply:
It does not store state in the index struct, does not support build/insert/update/delete, and does not expose a retriever.

Reply:
One possibility is that we only add this as query engine.

Reply:
cc @jerryjliu

Reply:
Calling it something else would help with some common misconceptions. Lots of people try to call `persist` on the SQL or Pandas indexes, which doesn't actually work